docs/design/structured-concurrency.md

```diff
@@ -0,0 +1,63 @@
+# Structured Concurrency
+
+This document gives an overview of the [CoroutineScope](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-coroutine-scope/)(s) 
+used by the SDK. It is meant to be a living document and provide additional context and guidance for SDK authors.
+
+## Definitions
+
+* **Coroutine** - An instance of a suspendable computation. A coroutine itself is represented by a Job. It is responsible for coroutine’s lifecycle, cancellation, and parent-child relations.
+* **CoroutineContext** - Persistent (immutable) context for a coroutine
+* **CoroutineScope** - Defines a scope for new coroutines which delimits the lifetime of the coroutine
+* **Structured Concurrency** - A principle expected by coroutines (in Kotlin) whereby new coroutines can only be launched into a specific CoroutineScope. This constraint ensures that coroutines are not leaked and that errors are propagated correctly. An outer scope cannot complete until all its child coroutines complete. 
+* **HttpCall** - A single HTTP request/response pair. 
+* **HttpClientEngine** - A component responsible for executing an HTTP request and returning an HttpCall
+* **ExecutionContext** - Operation scoped (mutable) context used to drive an operation to completion
+
+
+## Scopes
+
+The SDK has three implementations of `CoroutineScope`:
+
+1. [ExecutionContext](https://github.com/awslabs/smithy-kotlin/blob/main/runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/operation/ExecutionContext.kt#L16)
+2. [HttpClientEngine](https://github.com/awslabs/smithy-kotlin/blob/main/runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/engine/HttpClientEngine.kt#L20)
+3. [HttpCall](https://github.com/awslabs/smithy-kotlin/blob/main/runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/response/HttpCall.kt#L28)
```

Comment:
**Suggestion**: These aren't permalinks so their accuracy may degrade over time. Consider either removing the line number fragment or using permalinks.